### PR TITLE
a few warnings fixes

### PR DESCRIPTION
**Pull request by nikitatrifan**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -17,56 +15,55 @@ import { Elastic } from "./Routes/Projects/elastic";
...
 import { Plans } from "./Routes/Projects/plans";
 import { Library } from "./Routes/Projects/library";
+import { Guidelines } from "./Routes/guidelines";
 import { GridVisibilityProvider } from "./Hooks/useGridVisibility";
 
 function App() {
   return (
     <BrowserRouter basename={process.env.PUBLIC_URL + "/"}>
       <GridVisibilityProvider>
-        <Analytics id="UA-162053346-1" debug>
```
Comment (nikitatrifan) [inline src/App.js:29]: lets have it integrated later

File: src/Components/cardsLayout.js [hunk 2]
```diff
@@ -235,9 +235,12 @@ export function FooterCards({
                 titleColor={previousColor}
                 background={previousBackground}
                 backgroundHover={previousHover}
+                //
+                // note
+                //
                 alt
                 src
```
Comment (nikitatrifan) [inline src/Components/cardsLayout.js:242]: @dnvt hey bud, when an attribute in react is being used without any passed value, react will consider you want to pass a boolean `true` and as I can see you don't really expect such behavior here. It messes around with your router hehe

File: src/Components/cardsLayout.js [hunk 3]
```diff
@@ -260,7 +263,7 @@ export function FooterCards({
                 backgroundHover={nextHover}
                 alt
                 src
-                path
+                path="/"
```
Comment (nikitatrifan) [inline src/Components/cardsLayout.js:266]: you can just remove those empty attributes that are being `true` eventually so your components would apply default values on their own

File: src/Components/header.js [hunk 4]
```diff
@@ -42,45 +42,46 @@ export function Header(props) {
...
               value={"François Denavaut"}
               fill={colors.text}
-            ></MenuLink>
+            />
             <MenuLinkIcon
               path={"/"}
               value={"Work"}
               iconName={"Mouse"}
               fill={colors.text}
-            ></MenuLinkIcon>
```
Comment (nikitatrifan) [inline src/Components/header.js:51]: if there is no children provided inside an jsx element react throws a warning in the console saying blah blah blah you need to close tags

File: src/Components/hover.js [hunk 5]
```diff
@@ -7,20 +7,26 @@ export default function useHover() {
...
   useEffect(() => {
-    ref.current.addEventListener("mouseenter", enter);
-    ref.current.addEventListener("mouseleave", leave);
+    const currentRef = ref.current;
+
+    // Event handlers
+    const enter = () => setHovered(true);
+    const leave = () => setHovered(false);
+
+    if (!currentRef) {
```
Comment (nikitatrifan) [inline src/Components/hover.js:18]: @dnvt sometimes when you use ref to grab a node references, components where you pass it can just ignore them, like react-router-dom Link components and therefore there is nothing to attach your events. I added a little warning just in case.

File: src/Components/links.js [hunk 6]
```diff
@@ -191,24 +191,22 @@ export function Separator(props) {
...
-export function MenuLink({ path, value, fill, ...props }) {
+export function MenuLink({ path = '/', value, fill, ...props }) {
   const classes = linkStyles(props);
   const [ref, hovered] = useHover();
 
   return (
     <>
       <Link
-        ref={ref}
+        innerRef={ref}
```
Comment (nikitatrifan) [inline src/Components/links.js:205]: @dnvt react-router-dom for some reason has no support for direct Refs, so they have another attribute instead called `innerRef`. That way your useHover should be working fine.

File: src/Components/links.js [hunk 7]
```diff
@@ -226,11 +224,11 @@ export function MenuLinkIcon({ path, href, value, fill, iconName, ...props }) {
   const classes = linkStyles(props);
   const [ref, hovered] = useHover();
 
-  if ({ path }) {
+  if (path) {
```
Comment (nikitatrifan) [inline src/Components/links.js:227]: @dnvt if you try to have this returned into the console `({ path })` you'll get an object, like `{ path: undefined }` in case if you haven't passed any value into the props. So, here Javascript will convert your object into a boolean value to figure out if the body of the condition should be executed. Objects are always going to be converted to true, doesn't matter what they have inside. `if (value) {}` is exactly like `if (Boolean(value)) {}`

File: src/Components/links.js [hunk 8]
```diff
@@ -226,11 +224,11 @@ export function MenuLinkIcon({ path, href, value, fill, iconName, ...props }) {
...
     return (
       <>
         <Link
-          ref={ref}
+          innerRef={ref}
           className={classes.menuIcon}
           to={path}
           scroll={(el) =>
```
Comment (nikitatrifan) [inline src/Components/links.js:234]: btw I'm not sure if this prop works here. why is this here?

File: src/Components/links.js (hunk 7 above)
Comment (dnvt) [reply src/Components/links.js:227]: Makes a lot of sense ahah. Thanks

File: src/Components/links.js (hunk 8 above)
Comment (dnvt) [reply src/Components/links.js:234]: Ya now seen under that light it looks weird to me too. I was trying to use one component for two types of links
if path ? <Link to + scrollTo/> : <a href/>

File: src/Routes/Homepage/AboutSection.js [hunk 9]
```diff
@@ -60,26 +60,26 @@ export function AboutSection({ visibility }) {
   );
 }
 
-function FontimationPlacement({ x, y }) {
+function FontimationPlacement() {
```
Comment (dnvt) [inline src/Routes/Homepage/AboutSection.js:63]: I don't understand why you can get rid of the properties here? I can't see any Hook API?

File: src/index.js [hunk 10]
```diff
@@ -5,9 +5,7 @@ import App from './App';
 import * as serviceWorker from './serviceWorker';
 
 ReactDOM.render(
-  <React.StrictMode>
-    <App />
-  </React.StrictMode>,
```
Comment (dnvt) [inline src/index.js:10]: Are these equivalent to "use strict"; in pure js? Why do you feel it's better without? :)

File: src/Components/hover.js [hunk 11]
```diff
@@ -7,20 +7,26 @@ export default function useHover() {
...
 
-  // Event handlers
-  const enter = () => setHovered(true);
-  const leave = () => setHovered(false);
-
   // Simple effect, just bind and unbind the event handlers
   useEffect(() => {
-    ref.current.addEventListener("mouseenter", enter);
-    ref.current.addEventListener("mouseleave", leave);
+    const currentRef = ref.current;
```
Comment (dnvt) [inline src/Components/hover.js:12]: You can grab the actual ref with .current? Is it some type of .this for hooks?

File: src/Routes/Homepage/AboutSection.js (hunk 9 above)
Comment (nikitatrifan) [reply src/Routes/Homepage/AboutSection.js:63]: I discovered that even though x and y values were provided down the three to a FontimationPlacement component -- they were not used and, frankly, it seemed to me the component Fontimation doesnt need those values either. Have you tested if it works that way at all?

File: src/index.js (hunk 10 above)
Comment (nikitatrifan) [reply src/index.js:10]: Hm tbh I don't remember why I removed that. Wanna get it back?

File: src/Components/hover.js (hunk 11 above)
Comment (nikitatrifan) [reply src/Components/hover.js:12]: ref is basically a reference to an object current, that is immutable. meaning, if you change ref.current no re-render will be caused. for example you cant do that with useState. As an example:
```
let [val, setVal] = useState(1)
useEffect(() => {
   // nothing wi
  val = 0;
}, [setVal])

// val wont be updated at all if you try to mutate/redefine it
// as it will be wiped out during the next render, if anything gets updated
console.log(val); // 1

// but if you do it this way
useEffect(() => {
  setVal(0)
}, [setVal])
// val will get updated to 0
// and the component will be rerendered
// cuz you updated state
console.log(val); // will print out 1 during the first render, then 0 after the useEffect will get executed

// in the mean time, if you want to change something
// but you dont want to cause your component rerender everytime your value changes, use Ref
const formRef = useRef({});

async function handleSubmitForm() {
  const { username, password } = storeRef.current;
  await axios.get('/api', { username, password });
  alert('you got it!')
}

return (
  <form onSubmit={handleSubmitForm}>
     <input onChange={e => formRef.current.username = e.target.value} placeholder='' />
     <input onChange={e => formRef.current.password = e.target.value} placeholder='' />
  </form>
);

// that way it wont cause any re-renders
// as you just mutate ref and you'll have the latest
// data inside the submit callback no mater what
```

Comment (nikitatrifan) [reply src/Components/hover.js:12]: shiet this made me realize how complex react became over time since I first discovered it
